shm/MapAlternativeVersion.hpp

```diff
@@ -0,0 +1,16 @@
+#pragma once
+
+#include <vector>
+
+#include "Coordinates.hpp"
+#include "Island.hpp"
+
+class Map {
+    std::vector<Island> islands_;
+    Island* currentPosition_;
+
+public:
+    Map();
+    Island* getCurrentPosition() const;
+    Island* getIsland(const Coordinates&);
+};
```

Comment:
Good idea will be add `.clang-format` to project, so you can forget easliy about comments regarding to EOF's and format inconsistencies :)